src/pages/MissingPersonalDetails/MissingPersonalDetailsContent.tsx

```diff
@@ -0,0 +1,120 @@
+
+function MissingPersonalDetailsContent({privatePersonalDetails, cardList, draftValues}: MissingPersonalDetailsContentProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const ref: ForwardedRef<InteractiveStepSubHeaderHandle> = useRef(null);
+
+    const values = useMemo(() => getSubstepValues(privatePersonalDetails, draftValues), [privatePersonalDetails, draftValues]);
+
+    const startFrom = useMemo(() => getInitialSubstep(values), [values]);
+
+    const firstUnissuedCard = useMemo(() => Object.values(cardList ?? {}).find((card) => card.state === CONST.EXPENSIFY_CARD.STATE.STATE_NOT_ISSUED), [cardList]);
+
+    const handleFinishStep = useCallback(() => {
+        if (!values) {
+            return;
+        }
+        PersonalDetails.updatePersonalDetailsAndShipExpensifyCard(values, firstUnissuedCard?.cardID ?? 0);
+        FormActions.clearDraftValues(ONYXKEYS.FORMS.PERSONAL_DETAILS_FORM);
+        Navigation.goBack();
+    }, [firstUnissuedCard?.cardID, values]);
+
+    const {
+        componentToRender: SubStep,
+        isEditing,
+        nextScreen,
+        prevScreen,
+        screenIndex,
+        moveTo,
+        goToTheLastStep,
+    } = useSubStep<CustomSubStepProps>({bodyContent: formSteps, startFrom, onFinished: handleFinishStep});
+
+    const handleBackButtonPress = () => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+
+        // Clicking back on the first screen should dismiss the modal
+        if (screenIndex === CONST.MISSING_PERSONAL_DETAILS_INDEXES.MAPPING.LEGAL_NAME) {
+            FormActions.clearDraftValues(ONYXKEYS.FORMS.PERSONAL_DETAILS_FORM);
+            Navigation.goBack();
+            return;
+        }
+        ref.current?.movePrevious();
+        prevScreen();
+    };
+
+    const handleNextScreen = useCallback(() => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+        ref.current?.moveNext();
+        nextScreen();
+    }, [goToTheLastStep, isEditing, nextScreen]);
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            shouldEnableMaxHeight
+            testID={MissingPersonalDetailsContent.displayName}
+        >
+            <HeaderWithBackButton
+                title={translate('workspace.expensifyCard.addShippingDetails')}
+                onBackButtonPress={handleBackButtonPress}
+            />
+            <View style={[styles.ph5, styles.mb3, styles.mt3, {height: CONST.NETSUITE_FORM_STEPS_HEADER_HEIGHT}]}>
+                <InteractiveStepSubHeader
+                    ref={ref}
+                    startStepIndex={startFrom}
+                    stepNames={CONST.MISSING_PERSONAL_DETAILS_INDEXES.INDEX_LIST}
+                />
+            </View>
+            <SubStep
+                isEditing={isEditing}
+                onNext={handleNextScreen}
+                onMove={moveTo}
```

Comment:
We shouldn't have passed the `moveTo` callback as is because this logic is missing updating the `InteractiveStepSubHeader`. We should have created a wrapper similar to `handleNextScreen`.

This caused the `InteractiveStepSubHeader` to be out of sync https://github.com/Expensify/App/issues/56197